github/GithubIntegration.py

```diff
@@ -212,18 +218,34 @@ def create_jwt(self, expiration: int | None = None) -> str:
         return self.auth.create_jwt(expiration)
 
     def get_access_token(
-        self, installation_id: int, permissions: dict[str, str] | None = None
+        self,
+        installation_id: int,
+        permissions: dict[str, str] | None = None,
+        repositories: list[str | int] | None = None,
     ) -> InstallationAuthorization:
         """
         :calls: `POST /app/installations/{installation_id}/access_tokens <https://docs.github.com/en/rest/apps/apps#create-an-installation-access-token-for-an-app>`
         """
         if permissions is None:
             permissions = {}
 
+        if repositories is not None and not (isinstance(repositories, list) and all(isinstance(r, (str, int)) for r in repositories)):
```

Comment:
The unit test fails here because in Python, booleans are subclass of integers 🙃 

https://docs.python.org/3/c-api/bool.html

We could ether be explicit about the type check (exclude subclasses..)

```suggestion
        if repositories is not None and not (isinstance(repositories, list) and all(type(r) in (str, int) for r in repositories)):
```

Or change the unit test. Not sure why someone would pass in a subclassed object or str or int into the list of repositories but I will leave it up to you!

Reply:
Let's stick to `isinstance` as we use it everywhere while it is not accurate, as you pointed out. Can you fix the unittest by simply using `object()`?

Reply:
Makes sense, updated the test.